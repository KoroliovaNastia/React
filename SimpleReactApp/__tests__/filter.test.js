import React from 'react';
import { shallow, configure, mount, render } from 'enzyme';
import FilterButton from '../src/components/filterButton'
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('FilterButton component', () =>{
    const data = {text: "test", checked: false}
    it('should be call toggleChange', () => {
        const toggleChange = jest.fn();
        const component = mount(<FilterButton data={data} toggleChange={toggleChange}/>);
        const checkbox = () => component.find('input');
       // expect(checkbox()).toHaveProperty("checked", false);
       // checkbox().should.not.be.checked();
        checkbox().simulate('change', {target: {checked: true}});
        expect(toggleChange).toHaveBeenCalled();
        //expect(checkbox()).toHaveProperty("checked", true);
    })
})