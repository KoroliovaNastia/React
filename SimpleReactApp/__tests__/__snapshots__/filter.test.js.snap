// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter component should render correctly 1`] = `
<ContextProvider
  value={null}
>
  <Filter
    buttons={
      Array [
        Object {
          "checked": true,
          "field": "title",
          "id": "b1",
          "text": "title",
        },
        Object {
          "checked": false,
          "field": "genres",
          "id": "b2",
          "text": "genre",
        },
      ]
    }
    navigationFilters={
      Object {
        "buttonList": Array [
          Object {
            "checked": true,
            "field": "release_date",
            "id": "f1",
            "text": "Release date",
          },
          Object {
            "checked": false,
            "field": "vote_average",
            "id": "f2",
            "text": "rating",
          },
        ],
        "title": "sort by",
        "type": "CHANGE_NAVIGATION_FILTERS",
      }
    }
    query=""
    searchFilters={
      Object {
        "buttonList": Array [
          Object {
            "checked": true,
            "field": "title",
            "id": "b1",
            "text": "title",
          },
          Object {
            "checked": false,
            "field": "genres",
            "id": "b2",
            "text": "genre",
          },
        ],
        "title": "search by",
        "type": "CHANGE_SEARCH_FILTERS",
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    title="search by"
    type="CHANGE_SEARCH_FILTERS"
    updateFilterButtons={[Function]}
  />
</ContextProvider>
`;
